src/content/en/updates/2019/08/barcode-detection.md

```diff
@@ -0,0 +1,140 @@
+[capabilities project](/web/updates/capabilities). Following a successful Origin Trial,
+barcode detection has now launched.
+
+You may remember [my post from January](../01/shape-detection)
+on the [Shape Detection API](https://wicg.github.io/shape-detection-api),
+an API that defines three interfaces: `BarcodeDetector`, `FaceDetector`,
+and `TextDetector`. I'm now happy to share that following a successful Origin Trial,
+we're shipping the `BarcodeDetector` part of the Shape Detection API in Chrome&nbsp;78.
+The `FaceDetector` and the `TextDetector` interface still need some more fine-tuning.
+
+### Working With the `BarcodeDetector` {: #barcodedetector}
+
+Let's briefly recall how working with the `BarcodeDetector` looks like:
+
+```js
+const barcodeDetector = new BarcodeDetector();
+try {
+  const barcodes = await barcodeDetector.detect(image);
+  barcodes.forEach(barcode => console.log(barcode));
+} catch (err) {
+  console.error('Barcode detection failed:', err);
+}
+```
+
+## Current Status {: #status }
+
+| Step                                       | Status                               |
+| ------------------------------------------ | ------------------------------------ |
+| 1. Create explainer                        | [Complete][explainer]                |
+| 2. Create initial draft of specification   | [Complete][spec]                     |
+| 3. Gather feedback & iterate on design     | [Complete][wicg-discourse]           |
+| 4. Origin trial                            | [Complete][ot]                       |
+| 5. **Launch**                              | **Complete**                         |
+
+### Origin Trial Feedback Summary
+
+The primary feedback during the Origin Trial was around the limited availability
+of barcode detection support across the various Blink platforms.
+In response to this feedback, we've now added a new static `getSupportedFormats()` method
+to more easily allow developers to detect when the current platform supports
+the desired formats, and when a polyfill is necessary.
+We've also hooked up the `formats` hint options parameter to the underlying implementation,
+which allows it to optimize for detecting only the formats specified.
+The code sample below shows how the new method could be used to detect QR codes
+(among several
+[other possible barcode formats](https://wicg.github.io/shape-detection-api/#enumdef-barcodeformat)).
+
+```js
+// 🆕 Check if the platform supports QR codes.
+if ((await BarcodeDetector.getSupportedFormats()).includes('qr_code')) {
+  // Initialize the barcode detector. As this could be costly,
+  // do this only once and from thereon reuse the instance.
+  const barcodeDetector = new BarcodeDetector({
+    // Hint at the format(s) we're after, allowing the implementation
+    // to optimize and only run detection for the mentioned format(s).
+    formats: ['qr_code']
+  });
+  try {
+    // Try to decode QR codes.
```

Comment:
The name of the method tells us this. Why repeat it?

Reply:
Removed :-)